core/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -31,6 +34,12 @@
  */
 public class DeprecationLogger {
 
+    private final Logger logger;
+
+    public Logger getLogger() {
```

Comment:
It is probably worth documenting why it is ok to let this thing leak from inside `DeprecationLogger`. I feel like we shouldn't let it be public at all so users have to go through the special purpose deprecation logging methods so we can curate how they are used and what they do more carefully.